compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -1880,9 +1882,9 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
             LocalKind::Var | LocalKind::Temp => {}
         }
 
-        // When `#![feature(unsized_locals)]` is enabled, only function calls
+        // When `unsized_fn_params` or `unsized_locals` is enabled, only function calls
         // and nullary ops are checked in `check_call_dest`.
-        if !self.tcx().features().unsized_locals {
+        if !(self.tcx().features().unsized_locals || self.tcx().features().unsized_fn_params) {
```

Comment:
FWIW, I am a bit worried about how fragile this is, relying on 3 locations to use the same boolean condition. Can you move that condition into a helper function that is called in all 3 places to prevent that code from diverging?